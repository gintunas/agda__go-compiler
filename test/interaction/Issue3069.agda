
record Σ (A : Set) (B : A → Set) : Set where
  constructor _,_
  field
    fst : A
    snd : B fst

postulate
  P : {X : Set} → X → Set
  H : Set
  R : H → Set
  x : H
  y : R x
  help : (f : H) (p : R f) → P {X = Σ H R} (f , p)

record Category : Set₁ where
  field
    Hom : Set
    id : Hom
    comp : Hom → Hom
    neut-l : P (comp id)
    neut-r : Hom

slice : Category
slice = λ where
  .Category.Hom → Σ H R
  .Category.id → (x , y)
  .Category.comp (f , p) → (f , p)
  .Category.neut-l → help x _
  .Category.neut-r → {!!}
