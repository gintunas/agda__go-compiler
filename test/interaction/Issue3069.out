(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Category.Hom (λ { .Category.Hom → Σ H R ; .Category.id → x , y ; .Category.comp (f , p) → f , p ; .Category.neut-l → help x (Σ.snd (Category.id .extendedlambda0)) ; .Category.neut-r → ?0 }) " nil)
((last . 1) . (agda2-goals-action '(0)))
